unittests/api_tests.cpp

```diff
@@ -1203,6 +1211,132 @@ BOOST_FIXTURE_TEST_CASE(transaction_tests, TESTER) { try {
 } FC_LOG_AND_RETHROW() }
 
+/*************************************************************************************
+ * verify subjective limit test case
+ *************************************************************************************/
+BOOST_AUTO_TEST_CASE(inline_action_subjective_limit) { try {
+   const uint32_t _4k = 4 * 1024;
+   tester chain(setup_policy::full, db_read_mode::SPECULATIVE, {_4k + 100}, {_4k + 1});
+   tester chain2(setup_policy::full, db_read_mode::SPECULATIVE, {_4k + 100}, {_4k});
+   signed_block_ptr block;
+   for (int n=0; n < 2; ++n) {
+      block = chain.produce_block();
+      chain2.push_block(block);
+   }
+   chain.create_account( N(testapi) );
+   for (int n=0; n < 2; ++n) {
+      block = chain.produce_block();
+      chain2.push_block(block);
+   }
+   chain.set_code( N(testapi), contracts::test_api_wasm() );
+   block = chain.produce_block();
+   chain2.push_block(block);
+
+   block = CALL_TEST_FUNCTION_WITH_BLOCK(chain, "test_transaction", "send_action_4k", {}).second;
+   chain2.push_block(block);
+   block = chain.produce_block();
+   chain2.push_block(block);
+
+} FC_LOG_AND_RETHROW() }
+
+/*************************************************************************************
+ * verify objective limit test case
+ *************************************************************************************/
+BOOST_AUTO_TEST_CASE(inline_action_objective_limit) { try {
+   const uint32_t _4k = 4 * 1024;
+   tester chain(setup_policy::full, db_read_mode::SPECULATIVE, {_4k}, {_4k - 1});
+   chain.produce_blocks(2);
+   chain.create_account( N(testapi) );
+   chain.produce_blocks(100);
+   chain.set_code( N(testapi), contracts::test_api_wasm() );
+   chain.produce_block();
+
+   chain.push_action(config::system_account_name, N(setpriv), config::system_account_name,  mutable_variant_object()
+         ("account", "testapi")
+         ("is_priv", 1));
+   chain.produce_block();
+
+   BOOST_CHECK_EXCEPTION(CALL_TEST_FUNCTION(chain, "test_transaction", "send_action_4k", {}), inline_action_too_big,
+                         [](const fc::exception& e) {
+                            return expect_assert_message(e, "inline action too big");
+                         }
+   );
+
+} FC_LOG_AND_RETHROW() }
+
+BOOST_AUTO_TEST_CASE(deferred_inline_action_subjective_limit_failure) { try {
+   const uint32_t _4k = 4 * 1024;
+   tester chain(setup_policy::full, db_read_mode::SPECULATIVE, {_4k + 100}, {_4k});
+   chain.produce_blocks(2);
+   chain.create_accounts( {N(testapi), N(testapi2), N(alice)} );
+   chain.set_code( N(testapi), contracts::test_api_wasm() );
+   chain.set_code( N(testapi2), contracts::test_api_wasm() );
+   chain.produce_block();
+
+   transaction_trace_ptr trace;
+   auto c = chain.control->applied_transaction.connect([&](std::tuple<const transaction_trace_ptr&, const packed_transaction_ptr&> x) {
+         auto& t = std::get<0>(x);
+         if (t->scheduled) { trace = t; }
```

Comment:
How pervasive are the deferred transactions in test_api.wasm?  We should probably be preferring inline actions in new tests if possible without rewriting all tests immediately.